File: src/main/java/net/lomeli/trophyslots/core/command/CommandSetSlots.java
package net.lomeli.trophyslots.core.command;

import net.minecraft.command.CommandBase;
import net.minecraft.command.ICommandSender;
import net.minecraft.entity.player.EntityPlayerMP;
import net.minecraft.util.ChatComponentText;
import net.minecraft.util.ChatComponentTranslation;
import net.minecraft.util.StatCollector;

import net.lomeli.trophyslots.TrophySlots;
import net.lomeli.trophyslots.core.SlotUtil;
import net.lomeli.trophyslots.core.network.MessageSlotsClient;

public class CommandSetSlots extends CommandBase {
    @Override
    public String getCommandName() {
        return "set-slots";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 2;
    }

    @Override
    public String getCommandUsage(ICommandSender sender) {
        return "command.trophyslots.set-slots.usage";
    }

    @Override
    public void processCommand(ICommandSender sender, String[] args) {
        if (args.length == 2 || args.length == 3) {
            EntityPlayerMP player = args.length == 2 ? getCommandSenderAsPlayer(sender) : getPlayer(sender, args[1]);
            int slots = args.length == 2 ? parseString(args[1]) : parseString(args[2]);
            if (player != null && (slots >= 0 && slots <= SlotUtil.getMaxSlots())) {
                SlotUtil.setSlotsUnlocked(player, slots);
                TrophySlots.packetHandler.sendTo(new MessageSlotsClient(slots), player);
                sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.set-slots.success"), player.getDisplayName(), slots)));
                if (!player.func_147099_x().hasAchievementUnlocked(TrophySlots.firstSlot) && player.func_147099_x().canUnlockAchievement(TrophySlots.firstSlot))
                    player.addStat(TrophySlots.firstSlot, 1);
                if (slots >= SlotUtil.getMaxSlots() && !player.func_147099_x().hasAchievementUnlocked(TrophySlots.maxCapcity) && player.func_147099_x().canUnlockAchievement(TrophySlots.maxCapcity))
                    player.addStat(TrophySlots.maxCapcity, 1);
            } else
                sender.addChatMessage(new ChatComponentText(String.format(StatCollector.translateToLocal("command.trophyslots.set-slots.error"), SlotUtil.getMaxSlots())));
        } else
            sender.addChatMessage(new ChatComponentTranslation(getCommandUsage(sender)));
    }

    private int parseString(String st) {
        try {
            return Integer.parseInt(st);
        } catch (Exception e) {
            return -1;
        }
    }
}

File: src/main/kotlin/net/lomeli/trophyslots/core/command/CommandSetSlots.kt
package net.lomeli.trophyslots.core.command

import net.lomeli.trophyslots.capabilities.slots.SlotManager
import net.minecraft.command.CommandBase
import net.minecraft.command.ICommandSender
import net.minecraft.entity.player.EntityPlayerMP
import net.minecraft.server.MinecraftServer
import net.minecraft.util.text.TextComponentString
import net.minecraft.util.text.TextComponentTranslation
import net.minecraft.util.text.translation.I18n

class CommandSetSlots : CommandBase() {
    override fun execute(server: MinecraftServer?, sender: ICommandSender?, args: Array<out String>?) {
        if (sender != null) {
            if (args != null && (args.size == 2 || args.size == 3)) {
                val player: EntityPlayerMP
                val slots: Int
                if (args.size == 3) {
                    slots = parseInt(args[2])
                    player = CommandBase.getPlayer(server, sender, args[1])
                } else {
                    slots = parseInt(args[1])
                    player = CommandBase.getCommandSenderAsPlayer(sender)
                }
                val slotInfo = SlotManager.getPlayerSlotInfo(player)!!
                if (player != null && (slots > 0 && slots <= slotInfo.getMaxSlots())) {
                    slotInfo.setSlots(slots)
                    SlotManager.updateClient(player, slotInfo)
                    sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.set_slots.success").format(player.displayName.unformattedText, slots)))
                } else
                    sender.sendMessage(TextComponentString(I18n.translateToLocal("command.trophyslots.set_slots.error").format(slotInfo.getMaxSlots())))
            } else
                sender.sendMessage(TextComponentTranslation(getUsage(sender)))
        }
    }

    override fun getName(): String? = "set-slots"

    override fun getUsage(sender: ICommandSender?): String? = "command.trophyslots.set_slots.usage"

    override fun getRequiredPermissionLevel(): Int = 2
}